docs/reference/modules/asset.md

```diff
@@ -0,0 +1,214 @@
+
+#### `extensions` function
+
+#### Example
+
+Functions in the `extensions` property can take an `options` parameter to allow several modules to pass options to an extension through the [`extensionOptions`](#extensionoptions) property.
+
+<AposCodeBlock>
+
+```javascript
+const path = require("path");
+
+module.exports = {
+// ...
+webpack: {
+ extensions: {
+   addAlias(options) {
+     return {
+       resolve: {
+         alias: {
+           ...(options.alias || {}),
+           Utilities: path.resolve(process.cwd(), './lib/utils')
+         }
+       }
+     };
+   }
+ }
+},
+// ...
+};
+
+```
+<template v-slot:caption>
+modules/my-module/index.js
+</template>
+</AposCodeBlock>
+
+### `extensionOptions`
+Any module can pass options to an existing webpack `extension` as long as it was defined as a function accepting an `options` object. The `extensionOptions` property is an object that contains sub-properties matching each extension to be modified - in the preceding example, `addAlias()`. The values for each sub-property can be simple objects, in which case they merge into the options object to be passed to that extension. They can also be functions, in which case they take an `options` object containing the options so far and must return a new object with any desired changes made.
+
+#### Example
+
+Extending the previous example with an additional alias using a simple object:
+
+<AposCodeBlock>
+
+```javascript
+module.exports = {
+// ...
+webpack: {
+ extensionOptions: {
+   addAlias: {
+       alias: {
+         Data: path.resolve(process.cwd(), './lib/data')
+       }
+     }
+   }
+ }
+},
+// …
+};
+```
+<template v-slot:caption>
+module/my-new-widget/index.js
+</template>
+</AposCodeBlock>
+
+### `bundles`
+The `bundles` option allows delivery of frontend assets to a specific page or if a widget is on a page page. The` bundles` subproperty can be present in any type of module, not just widget type and page type modules. However, configuring a bundle in a widget type or page type module is the only way to cause it to actually be loaded. If a bundle is present in another type of module, that module can contribute code to it, but it will only get loaded when also configured for at least one widget type or page type. This option takes an object containing either the JavaScript or stylesheet file names without extension as properties. These files should be located in the module's `ui/src` folder. Each property takes either an empty object or a sub-property of `templates`. The `templates` sub-property accepts an array of page template names where the content should be loaded. The `templates` feature is most often relevant for piece-page-type modules; if it is not used, it is assumed both `index` and `show` templates should load the bundle.
```

Comment:
The space before "bundles" should be placed before the backtick.

Reply:
Changed.